router/router.nginx.conf.in

```diff
@@ -232,3 +232,20 @@ server {
     listen 80;
     listen [::]:80;
 }
+
+server {
+    server_name blog.*;
+
+    location / {
+        proxy_pass http://blog/;
+
+        proxy_set_header Host $updated_host;
+        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+        proxy_set_header X-Forwarded-Proto $updated_scheme;
```

Comment:
Do either of you know where `update_host` and `updated_scheme` come from? I've never seen that before. I've always used `$host` and `$scheme`. I can't find any references to these variables on the internet.

Reply:
Yeah, it's totally something we created. I think it may have been inspired by Kubernetes use of it. 

It's at the top of router.nginx.conf.in. The name is probably not quite right: it refers to the original host (as set by the upstream server in X-Forwarded-Host, X-Forwarded-Proto)

Reply:
I think the reason cotton used this is that by default aiohttp doesn't support X-Forwarded-Host (there is a module, aiohttp-remote that addresses this), and so he stuffs that into Host, falling back to $http_host if $http_x_forwarded_host is empty.

Reply:
This create a polymorphism that adds a bit of complexity, and potential source of bugs (if X-Forwarded-Host has multiple values, which is allowed by the spec), while also saving some complexity (not needing to look at a 2nd header field).